js/packages/mobile-wallet-adapter-protocol-web3js/src/transact.ts

```diff
@@ -158,3 +178,18 @@ export async function transact<TReturn>(
     };
     return await baseTransact(augmentedCallback, config);
 }
+
+function validateSecureEndpointSpecificURI(walletUriBase?: string) {
+    if (walletUriBase) {
+        let url: URL | null = null;
+        try {
+            url = new URL(walletUriBase);
```

Comment:
Fun fact: React Native doesn't offer the `URL` class. You have to polyfill it. Annoying.